From 2246085ae554cea03862710d084a16b0fa9e3a49 Mon Sep 17 00:00:00 2001
From: "K.Takata" <kentkt@csc.jp>
Date: Tue, 9 Oct 2018 16:41:14 +0900
Subject: [PATCH 33/33] dosinst: Remove needless sleeps

EnumWindows() is not an asynchronous function. No need to sleep after
the function.

No need to wait 3 seconds after showing "Done!" when the stdout is not
a tty.
---
 src/dosinst.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/dosinst.c b/src/dosinst.c
index 8b8586a19..6c98fc71f 100644
--- a/src/dosinst.c
+++ b/src/dosinst.c
@@ -18,6 +18,7 @@
  */
 #define DOSINST
 #include "dosinst.h"
+#include <io.h>
 
 #define GVIMEXT64_PATH	    "GvimExt64\\gvimext.dll"
 #define GVIMEXT32_PATH	    "GvimExt32\\gvimext.dll"
@@ -569,7 +570,6 @@ uninstall_check(int skip_question)
 			sleep(1);  /* wait for uninstaller to start up */
 			num_windows = 0;
 			EnumWindows(window_cb, 0);
-			sleep(1);  /* wait for windows to be counted */
 			if (num_windows == 0)
 			{
 			    /* Did not find the uninstaller, ask user to press
@@ -585,9 +585,9 @@ uninstall_check(int skip_question)
 			    {
 				printf(".");
 				fflush(stdout);
+				sleep(1);  /* wait for the uninstaller to finish */
 				num_windows = 0;
 				EnumWindows(window_cb, 0);
-				sleep(1);  /* wait for windows to be counted */
 			    } while (num_windows > 0);
 			}
 			printf("\nDone!\n");
@@ -2592,7 +2592,7 @@ main(int argc, char **argv)
 
 	/* When nothing found exit quietly.  If something found wait for
 	 * a little while, so that the user can read the messages. */
-	if (i)
+	if (i && _isatty(1))
 	    sleep(3);
 	exit(0);
     }
-- 
2.17.0

